src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -791,13 +791,17 @@ private int RunCore(TextWriter consoleOutput, ErrorLogger errorLogger, Cancellat
             // We pass it to the generators, which will realize any symbols they require. 
             compilation = RunGenerators(compilation, Arguments.ParseOptions, generators, additionalTexts, diagnostics);
 
-            // https://github.com/dotnet/roslyn/issues/44087 
```

Comment:
Why did we remove this comment? Feel like this code still needs to be cleaned up once we have a way to determine if `SyntaxTree` instances are generated or not.

Reply:
So, I started down that path, but it turns out we don't actually need it. @agocke work removing the diagnostics from the syntax tree means we won't have to parse with the config present. Because the syntax trees are deterministically ordered, we should always just be able to count beyond the length of the `Arguments.SyntaxTrees` to know which were added by generators?

---
In reply to: [434196389](https://github.com/dotnet/roslyn/pull/44768#discussion_r434196389) [](ancestors = 434196389)

Reply:
Admittedly, It feels like there is a lot of 'spaghetti' tangle in the way we handle this, various different levels of the compiler adjust the embedded texts and the config options based on their own rules and knowledge of what's coming later. I don't particularly like it, and it feels ripe for a refactor/cleanup pass but this PR at least doesn't make it _worse_. 

---
In reply to: [434210846](https://github.com/dotnet/roslyn/pull/44768#discussion_r434210846) [](ancestors = 434210846,434196389)

Reply:
K. Have we checked with @jasonmalinowski on whether or not he feels the IDE team will need to be able to recognize generated `SyntaxTree` instances over non-generated ones?

Reply:
We'll almost definitely need to do it in the future, its just not needed for this particular feature. (The IDE already works with the embedded texts and doesn't need to know they're generated)

---
In reply to: [434221150](https://github.com/dotnet/roslyn/pull/44768#discussion_r434221150) [](ancestors = 434221150)